src/output.ts

```diff
@@ -46,9 +46,9 @@ export class TSOutput {
   }
 
   public static async create(genPath) {
-    const st = await stat(genPath);
-    if (!st.isDirectory()) {
-      throw new Error(`output dir: ${genPath} is not a directory`);
+    const dirExists = await exists(genPath);
+    if (dirExists) {
```

Comment:
If you `mkdir`, don't check whether the directory exists -- it will fail.  This code probably behaves strangely with symlinks.